src/benchmarks.zig

```diff
@@ -0,0 +1,272 @@
+const std = @import("std");
+const Decl = std.builtin.Type.Declaration;
+
+const debug = std.debug;
+const io = std.io;
+const math = std.math;
+const mem = std.mem;
+const meta = std.meta;
+const time = std.time;
+
+/// to run gossip benchmarks:
+/// zig build benchmark -- gossip
+pub fn main() !void {
+    var allocator = std.heap.page_allocator;
+    var cli_args = try std.process.argsWithAllocator(allocator);
+    defer cli_args.deinit();
+
+    _ = cli_args.skip();
+    var maybe_filter = cli_args.next();
+    var filter = blk: {
+        if (maybe_filter) |filter| {
+            std.debug.print("filtering benchmarks with prefix: {s}\n", .{filter});
+            break :blk filter;
+        } else {
+            std.debug.print("no filter: running all benchmarks\n", .{});
+            break :blk "";
+        }
+    };
+
+    // TODO: very manual for now (bc we only have 2 benchmarks)
+    // if we have more benchmarks we can make this more efficient
+    const max_time_per_bench = 500; // !!
+
+    if (std.mem.startsWith(u8, "socket_utils", filter)) {
+        try benchmark(
+            @import("gossip/socket_utils.zig").BenchmarkPacketProcessing,
+            max_time_per_bench,
+            TimeUnits.milliseconds,
+        );
+    }
+
+    if (std.mem.startsWith(u8, "gossip", filter)) {
+        try benchmark(
+            @import("gossip/gossip_service.zig").BenchmarkMessageProcessing,
+            max_time_per_bench,
+            TimeUnits.milliseconds,
```

Comment:
I think it would be nice to have the benchmarking auto select the time units (by repeated dividing by 1000 to determine magnitude of nanoseconds) rather than explicitly passing it in - not a big deal either way but might be nice as the number of benchmarks grow.